qiskit_dynamics/backend/dynamics_backend.py

```diff
@@ -182,14 +194,20 @@ def set_options(self, **fields):
                         """initial_state must be either "ground_state",
                         or a Statevector or DensityMatrix instance."""
                     )
-            elif key == "meas_level" and value != 2:
-                raise QiskitError("Only meas_level == 2 is supported by DynamicsBackend.")
+            elif key == "meas_level" and value not in [1, 2]:
+                raise QiskitError("Only meas_level 1 and 2 are supported by DynamicsBackend.")
+            elif key == "meas_return" and value not in ["single", "avg"]:
+                raise QiskitError("meas_return must be either 'single' or 'avg'.")
             elif key == "max_outcome_level":
                 if (value is not None) and (not isinstance(value, int) or (value <= 0)):
                     raise QiskitError("max_outcome_level must be a positive integer or None.")
             elif key == "experiment_result_function" and not callable(value):
                 raise QiskitError("experiment_result_function must be callable.")
-
+            elif key == "iq_width" and (not isinstance(value, float) or (value <= 0)):
+                raise QiskitError("iq_width must be positive float.")
+            elif key == "iq_centers":
```

Comment:
Why should we have `validate_iq_centers  = True` in the `elif` block of `subsystem_dims` and then ensure it works for `iq_centers =None`? 
What use case are we trying to convert here?

Reply:
Changing it to an array might not work as the length for each subsystem will be different, creating an array of `dtype=object`

Reply:
I am thinking something like this:

```python
elif key == "iq_centers":
    validate_iq_centers = True
    self._options.update_options(**{key: value})
```
then:

```python
if validate_iq_centers:
    iq_centers = self._options.iq_centers
    if ([len(x) for x in iq_centers] != self._options.subsystem_dims) or (
        np.concatenate(iq_centers).shape[1] != 2
    ):
        raise QiskitError("iq_centers is not consistent with subsystem_dims.")
```

Reply:
> Changing it to an array might not work as the length for each subsystem will be different, creating an array of `dtype=object`

Ah yeah true, good point.

Reply:
This block:
```
elif key == "iq_centers":
    validate_iq_centers = True
    self._options.update_options(**{key: value})
```
Should need the `self._options.update_options(**{key: value})` as it will be handled by the general `else` of:
```
if key == "solver":
    self._set_solver(value)
    validate_subsystem_dims = True
else:
    if key == "subsystem_dims":
        validate_subsystem_dims = True
    self._options.update_options(**{key: value})
```

Once the `if` statement for `key == "subsystem_dims"` is moved to the upper if block. I think also we should also move the line `validate_subsystem_dims = True` into the upper block by adding a `elif key == "solver"` to that section.

Sorry I realize this is somewhat arbitrary, and this method is just validating and not doing any computation, but this nitpicking is in an attempt to clearly delineate what's happening in each code block of this method, so that it's easier to read in the future (since it's all validation/option setting, I have a feeling this could potentially get confusing in the future if we're not careful). What I'm thinking for the block of this method is: 

1. The first if block does immediate type and independent value validation (i.e. validating that the value of the option being set is well-formed), while also setting flags to do consistency validation later if the option needs to be consistent with other option values.
2. The second block sets the option - the main thing is the `else` block doing general option setting, but some options may have custom setting routines. In this case only the `solver` has a custom routine.
3. The final block does consistency validation based on which flags were raised in the first block.

Make the changes you want to make and try to follow the above - depending on how it looks, afterwards it may be more expedient for me to just open a PR to yours to make this method exactly how I want (since this really is mostly about future-me's ability to read it after forgetting the details :D ).

Reply:
This is almost there:

```
if validate_iq_centers:
    iq_centers = self._options.iq_centers
    if ([len(x) for x in iq_centers] != self._options.subsystem_dims):
        raise QiskitError("iq_centers is not consistent with subsystem_dims.")
    if not all([len(level)==2 for sub_system in iq_centers for level in sub_system]):
        raise QiskitError("iq_centers must be 2-tuples.")
```

changes:

1. Make sure `iq_centers is None` is accepted, so either envelope everything in another `if` statement that only does further validation if `iq_centers is not None`, or maybe add it to the overarching if statement, e.g. `if validate_iq_centers and self._options.iq_centers is not None`.
2. I think this `if` statement: `if not all([len(level)==2 for sub_system in iq_centers for level in sub_system])` needs to be moved into the first block as described above, since it's not part of a consistency check, it's just part of the check that `iq_centers` is in the `List[List[List[float,float]]]` format. You also need to be careful here that you're assuming you can call `len` on `level` here.

The first `if` statement is fine, assuming that you've already properly validated the `List[List[List[float,float]]]` format at this point.